src/local_scheduler/local_scheduler.cc

```diff
@@ -412,94 +412,106 @@ LocalSchedulerState *LocalSchedulerState_init(
 }
 
 /* TODO(atumanov): vectorize resource counts on input. */
-bool check_dynamic_resources(LocalSchedulerState *state,
-                             double num_cpus,
-                             double num_gpus,
-                             double num_custom_resource) {
-  if (num_cpus > 0 && state->dynamic_resources[ResourceIndex_CPU] < num_cpus) {
-    /* We only use this check when num_cpus is positive so that we can still
-     * create actors even when the CPUs are oversubscribed. */
-    return false;
-  }
-  if (num_custom_resource > 0 &&
-      state->dynamic_resources[ResourceIndex_CustomResource] <
-          num_custom_resource) {
-    return false;
-  }
-  if (state->dynamic_resources[ResourceIndex_GPU] < num_gpus) {
-    return false;
+bool check_dynamic_resources(
+    LocalSchedulerState *state,
+    const std::unordered_map<std::string, double> &resources) {
+  for (auto const &resource_pair : resources) {
```

Comment:
There's a slight change in intended usage here. Before, we only checked `CPU` resources if the passed in quantity was `> 0`. Now if we don't want to check `CPU` resources, then `CPU` should simply not be in `resources`.